pkgs/development/compilers/elm/default.nix

```diff
@@ -212,6 +212,32 @@ in lib.makeScope pkgs.newScope (self: with self; {
         };
       };
 
+      elm-pages = nodePkgs."elm-pages".overrideAttrs (
+        old: {
+          buildInputs = old.buildInputs ++ [ pkgs.makeWrapper ];
+
+          # can't use `patches = [ <patch_file> ]` with a nodePkgs derivation;
+          # need to patch in one of the build phases instead.
+          # see upstream issue https://github.com/dillonkearns/elm-pages/issues/305 for dealing with the read-only problem
+          preFixup = ''
+            patch $out/lib/node_modules/elm-pages/generator/src/codegen.js ${./packages/elm-pages-fix-read-only.patch}
+          '';
```

Comment:
```suggestion
          patches = [ ''./packages/elm-pages-fix-read-only.patch'' ];
```

Reply:
I did try this when I started this work, but this error appears in the build logs:
```
patching sources
applying patch /nix/store/zw27jx69p2m0a5a3cy89nk2rsknh9c5r-elm-pages-fix-read-only.patch
can't find file to patch at input line 5
Perhaps you used the wrong -p or --strip option?
The text leading up to this was:
--------------------------
|diff --git a/generator/src/codegen.js b/generator/src/codegen.js
|index a0ef8806e..fe00ac904 100644
|--- a/generator/src/codegen.js
|+++ b/generator/src/codegen.js
--------------------------
File to patch: 
Skip this patch? [y] 
Skipping patch.
1 out of 1 hunk ignored
```

Lemme poke at it a bit, will push a commit when I have `patches` working (or I give up)

Reply:
it might be the case that node2nix builder doesn't support this. I haven't tried it. In that case it would be fine to keep it as is.

Reply:
I believe there is even `patchPhase` but again I'm now not sure if it works with derivations from node2nix. I don't think postInstall is the right place for applying patches even though in this case it would be possible. I would try `patchPhase` and if that doesn't work I would keep it in `preFixup`

Reply:
`patchPhase` doesn't work either, seeing the following repeated ad infinitum:

```
       > patch: **** Can't create file /nix/store/nwk4ir8gxc2fc9ma3xg82vzgqf7dzkx5-elm-pages-2.1.11/lib/node_modules/elm-pages/generator/src/codegen.js.orig : Too many open files
       > patch: **** Can't create file /nix/store/nwk4ir8gxc2fc9ma3xg82vzgqf7dzkx5-elm-pages-2.1.11/lib/node_modules/elm-pages/generator/src/codegen.js.orig : Too many open files
       > patch: **** Can't create file /nix/store/nwk4ir8gxc2fc9ma3xg82vzgqf7dzkx5-elm-pages-2.1.11/lib/node_modules/elm-pages/generator/src/codegen.js.orig : Too many open files
```

Will leave it in `preFixup` as you've advised.